src/spaceone/inventory/service/collector_service.py

```diff
@@ -82,27 +82,44 @@ def collect(self, params):
             )
             yield error_resource_response.to_primitive()
 
+        # Execute manager (Multithreading)
+        # with concurrent.futures.ThreadPoolExecutor(max_workers=MAX_WORKER) as executor:
+        #     future_executors = []
+        #     for execute_manager in self.execute_managers:
+        #         _manager = self.locator.get_manager(execute_manager)
+        #         future_executors.append(
+        #             executor.submit(_manager.collect_resources, params)
+        #         )
+        #
+        #     for future in concurrent.futures.as_completed(future_executors):
+        #         try:
+        #             for result in future.result():
+        #                 yield result.to_primitive()
+        #         except Exception as e:
+        #             _LOGGER.error(
+        #                 f"[collect] failed to yield result => {e}", exc_info=True
+        #             )
+        #             error_resource_response = self.generate_error_response(
+        #                 e, "", "inventory.Error"
+        #             )
+        #             yield error_resource_response.to_primitive()
+
         # Execute manager
-        with concurrent.futures.ThreadPoolExecutor(max_workers=MAX_WORKER) as executor:
-            future_executors = []
-            for execute_manager in self.execute_managers:
-                _manager = self.locator.get_manager(execute_manager)
-                future_executors.append(
-                    executor.submit(_manager.collect_resources, params)
+        future_executors = []
+        for execute_manager in self.execute_managers:
+            _manager = self.locator.get_manager(execute_manager)
+            future_executors.append(_manager.collect_resources(params))
+
+        for future_executor in future_executors:
+            try:
+                for result in future_executor:
+                    yield result.to_primitive()
+            except Exception as e:
+                _LOGGER.error(f"[collect] failed to yield result => {e}", exc_info=True)
+                error_resource_response = self.generate_error_response(
+                    e, "", "inventory.Error"
                 )
-
-            for future in concurrent.futures.as_completed(future_executors):
-                try:
-                    for result in future.result():
-                        yield result.to_primitive()
-                except Exception as e:
-                    _LOGGER.error(
-                        f"[collect] failed to yield result => {e}", exc_info=True
-                    )
-                    error_resource_response = self.generate_error_response(
-                        e, "", "inventory.Error"
-                    )
-                    yield error_resource_response.to_primitive()
+                yield error_resource_response.to_primitive()
```

Comment:
I thought of 3 changes while looking at the code.

1. I think the part where the executor repeats twice is duplicated.
2. In the case of `_manager`, `_`seems to be used to prevent access to internal variables, but it is questionable to use `_` expressions in a single method except for self variable or class variable in class.
3. It seems better to set the error_resource from the try to exception part to the part where the manager is called. The reason is that I thought I could catch the error that could occur in `get_manager`.

Examples of codes reflecting three points are as follows.

```python
    for execute_manager in self.execute_managers:
    
        try:
            target_manager = self.locator.get_manager(execute_manager)
            future_executor = target_manager.collect_resources(params)
            for result in future_executor:
                yield result.to_primitive()
        except Exception as e:
            _LOGGER.error(f"[collect] failed to yield result => {e}", exc_info=True)
            error_resource_response = self.generate_error_response(
                e, "", "inventory.Error"
            )
            yield error_resource_response.to_primitive()
```

I'm curious about Youngjin's opinion on the above part. Thanks : )

Reply:
Confirmed that the new commit is reflected. Accordingly, there are two questions.

1. Does the changed structure work well locally?
2. The target_manager expression is not a good way in terms of clean code. Is there a better way?

Also, I would like to ask @yjinjo  if there were any parts that were a bit ambiguous while working.

Reply:
1. It worked well in a local environment
2. Well, I think it is a better way than what I wrote before. Since it is used only in a for loop, it is not necessary to use a protected variable like `_manager`  

One thing I would like to mention is I thought it is not necessary to wrap my code to the try-except clause again because this function was already checked from the upper function, but It is better than just for loop as you mentioned. 

Thank you for commenting. You helped me a lot. 👍